src/main/java/org/jenkinsci/plugins/codesonar/api/CodeSonarHubAnalysisDataLoader.java

```diff
@@ -0,0 +1,238 @@
+package org.jenkinsci.plugins.codesonar.api;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.jenkinsci.plugins.codesonar.CodeSonarAlertCounter;
+import org.jenkinsci.plugins.codesonar.CodeSonarAlertLevels;
+import org.jenkinsci.plugins.codesonar.models.CodeSonarHubInfo;
+import org.jenkinsci.plugins.codesonar.models.Metric;
+import org.jenkinsci.plugins.codesonar.models.ProcedureMetric;
+import org.jenkinsci.plugins.codesonar.models.analysis.Analysis;
+import org.jenkinsci.plugins.codesonar.models.analysis.Warning;
+import org.jenkinsci.plugins.codesonar.models.json.CodeSonarAnalysisWarningCount;
+import org.jenkinsci.plugins.codesonar.models.metrics.Metrics;
+import org.jenkinsci.plugins.codesonar.models.procedures.ProcedureRow;
+import org.jenkinsci.plugins.codesonar.models.procedures.Procedures;
+import org.jenkinsci.plugins.codesonar.services.HttpService;
+
+public class CodeSonarHubAnalysisDataLoader {
```

Comment:
I think this class needs a comment that explains the "legacy" vs. the current, JSON-API mechanisms.  For example

```
/** Loads and caches data related to a single analysis, for use by "Condition" classes.

    This class implements two distinct modes for loading data: a "legacy" mode, and the current mode.
    The legacy mode loads large lists of warnings and procedures into main memory.
    This was the only mode implemented through plugin version 3.3.x.
    The current mode fetches analysis data in a more memory-safe way, but it requires CodeSonar hub JSON APIs available only in CodeSonar 7.3 and later (i.e. with the "gridConfigJson" capability).
*/